File: ca_back/src/main/java/fr/sigma/ca/domain/Commission.java
package fr.sigma.ca.domain;

import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;
import java.sql.Timestamp;
import java.util.UUID;

@Entity
@Data
public class Commission {

        private static final long serialVersionUID = 1L;

        @Id
        private UUID id;

        @OneToOne
        private Negociateur negociateur;

        private BigDecimal pourcentage;

        private BigDecimal montantHT;

        private Timestamp dateVente;
}

File: ca_front/ca-front/src/app/commission/dto/commission.ts
import { Negociateur } from '../../negociateur/dto/negociateur';

export class Commission{

    pourcentage: number;
    montantHT: number;
    dateVente: Date;
    negociateur: Negociateur;

    constructor(
      pourcentage: number,
      montantHT: number,
      dateVente: Date = new Date(),
      negociateur: Negociateur
    ){
        this.pourcentage = pourcentage;
        this.montantHT = montantHT;
        this.dateVente = dateVente;
        this.negociateur = negociateur;
    }
}
